### ISSUE TITLE
Exported KMZ symbology does not match the source symbology (ArcMap)

### ISSUE THREAD

**Issue by BobBooth** [Esri/distance-direction-addin-dotnet#452]
I created some circles with the DD circle tool.
Save As...
Chose KMZ
Export.
Upload exported KMZ to AGOL.
Add to a map.
The resulting KMZ does not retain the symbology of the graphics (e.g. line width, color, no fill) that I created in ArcMap. I would expect the KMZ to have red circles in it, with at least approximately similar symbology and labels.
![image](https://user-images.githubusercontent.com/3801392/30932536-bffaf582-a395-11e7-9384-6eba4338316d.png)

Testing with ArcMap 10.6 and Distance and Direction Add-In 2.1.1

**Comment by BobBooth** [comment]
Likewise lines, ellipses, rings.

**Comment by csmoore** [comment]
The filled-circles/ellipses issue also seen in Pro 2.1

**Comment by BobBooth** [comment]
Not any better in ArcGIS Earth, so it is a problem with the Export functionality not setting the color, not AGOL mis-reading it.
![image](https://user-images.githubusercontent.com/3801392/31082235-6494b992-a75c-11e7-9994-bcc4862c806a.png)

**Comment by csmoore** [comment]
This makes sense since I don't think the symbology was being set on the exported layers. With the change to layers/feature classes in Pro, hopefully this will now be addressed in Pro (after #559 is complete) and assuming Conversion Tools.LayerToKML works as expected.

**Comment by csmoore** [comment]
The current state of this issue is:

**Pro:** - This should now be supported  - with the same limitations as the [LayerToKML GP Tool](http://pro.arcgis.com/en/pro-app/tool-reference/conversion/layer-to-kml.htm) - as noted in http://pro.arcgis.com/en/pro-app/tool-reference/conversion/kml-conversion.htm -  ex: 
*For line features, only simple symbols with basic color and line width properties are supported; advanced effects such as dashes and arrows and multilevel or layered symbols are not supported.*

**ArcMap** - There is currently no symbology being applied to the exported feature class prior to LayerToKML call - to add this support some changes would need to be made to either:
1. Create features in feature layers like is now being done in Pro
-or-  
1. Create Lyr files with symbology for each shape type 
2. Change the export process to apply this layer file to the exported feature class

**Comment by BobBooth** [comment]
[MakeLYR4KML.zip](https://github.com/Esri/distance-direction-addin-dotnet/files/2084900/MakeLYR4KML.zip)
@csmoore - zipfile of LYR files attached - see if that'll work.

**Comment by csmoore** [comment]
@BobBooth - thanks for putting these lyr files together for this issue.

This should be ready to work on whenever added to sprint (it currently shows June Release).

Given the current state we can probably **lower the estimate to a 3** - I'd make lower but I'm not entirely sure how  to apply the lyr files to a Map/TOC layer in ArcObjects/Engine (i.e. how to do what ArcMap does with the Import Symbology button - although there is a Map method to add layer from lyr file which might work if we add the layer 1st and then populate the data)

**Comment by dfoll** [comment]
@topowright @kgonzago @lfunkhouser build 12  in ....\Commercial\Solutions\DefenseSolutions\DistanceAndDirection\Testing can be used to test this functionality

**Comment by dfoll** [comment]
This is the symbology from ArcMap with build 12
![image](https://user-images.githubusercontent.com/5322669/46980841-bbd28380-d08a-11e8-918f-fb7ebfcd6c79.png)

If this is a limitation of ArcMap, we should mark this as won't fix and close.

**Comment by csmoore** [comment]
@dfoll here is the ArcMap help topic that mentions limitations in Layer to KML tool: 

http://desktop.arcgis.com/en/arcmap/10.5/manage-data/kml/creating-kml-in-arcgis-for-desktop.htm 

"Layer Symbology" row of this table: http://desktop.arcgis.com/en/arcmap/10.5/manage-data/kml/creating-kml-in-arcgis-for-desktop.htm#ESRI_SECTION1_1526893BE9924595BBEC586CAFC97B0B

**Comment by topowright** [comment]
@CTLocalGovTeam has came back with the approach that this cannot be fixed inside of ArcMap. They belive that it is a known limitation of ArcMap. Currently we are looking for documentation that supports this.

**Comment by dfoll** [comment]
http://desktop.arcgis.com/en/arcmap/10.3/manage-data/kml/creating-kml-in-arcgis-for-desktop.htm

States that simple symbology from a feature layer can be used for kml symbology. Do we need to implement something to save to a feature layer first instead of exporting directly to kml?

**Comment by BobBooth** [comment]
There is an example here that may be useful:
http://edndoc.esri.com/arcobjectsonline/componenthelp/esriCore/IUniqueValueRenderer_Example.htm

**Comment by dfoll** [comment]
@CTLocalGovTeam was this sample helpful in determining if there can be a way forward on this issue?

**Comment by lfunkhouser** [comment]
@saip and @CTLocalGovTeam please identify if the sample provided by @BobBooth can be used to move forward on this issue?

**Comment by dfoll** [comment]
spoke with @CTLocalGovTeam this morning. They do not have documentation that this is not supported, it just is not working for them. They have asked us to reach out to the core team to ask if we we can track down any further information about why this does not work.

**Comment by dfoll** [comment]
@CTLocalGovTeam I am revisiting this issue, trying to get this closed out. Unfortunately most discussions about the development on this issue took place before I assumed my current role. In this [comment](https://github.com/Esri/distance-direction-addin-dotnet/issues/452#issuecomment-395752459) @csmoore  proposed possible solution. Can it be confirmed that steps 2/3 do not work?

If steps 2/3 were not attempted, could we try that? 
@csmoore provided this additional info
_just like you set the output symbology on a gp tool - the API should allow the same thing_ 
He is also available if we need to discuss further

**Comment by csmoore** [comment]
@dfoll @saip 

Calling ApplySymbologyFromLayer with the layer files from @BobBooth [above](https://github.com/Esri/distance-direction-addin-dotnet/issues/452#issuecomment-395790115) seemed to work for me (I tested on ArcMap 10.6.1).

I made changes to support this in branch: https://github.com/Esri/distance-direction-addin-dotnet/tree/csm/%23452-export-kml-arcmap  -  in case you want to see if you can reproduce these results

![image](https://user-images.githubusercontent.com/3090809/49120863-7786f580-f27c-11e8-875c-dcb485a61ece.png)

**Comment by saip** [comment]
@dfoll @csmoore We have tried opening the exported kmz on https://www.google.com/maps and it does not retain the symbology(this was what we used when we tried last time). We have also tested it both on ArcGIS Earth and Google Earth and it looks good with the output generated from both 10.3.1 and 10.6.1 ArcMap versions.

Code changes and test results from the addin look good too in Google Earth and ArcGIS Earth.

**Comment by dfoll** [comment]
@csmoore can you please do a PR for this

**Comment by csmoore** [comment]
PR #638 created.

I do see the KML line styles not being retained in Google Maps - it looks like Google Maps data import doesn't import the line styles so good to know that we shouldn't use it to test - but we should be able to use ArcGIS Pro/Earth or Google Earth.

Google Maps Import:
![image](https://user-images.githubusercontent.com/3090809/49154071-7a6a0080-f2e5-11e8-87e3-9de07c2d160e.png)

**Comment by dfoll** [comment]
Build 15 was created from @csmoore branch, to test

**Comment by dfoll** [comment]
![image](https://user-images.githubusercontent.com/5322669/49229440-82946f80-f3bb-11e8-8ec1-2ae6ddf6dc9a.png)

Looks good when building from this [branch](https://github.com/Esri/distance-direction-addin-dotnet/tree/csm/%23452-export-kml-arcmap). @csmoore and @saip have also said they tested from this branch and looks good. Will merge to dev and build again.

**Comment by dfoll** [comment]
This was listed as a known issue in Sept 2018... For January 2019 it was already removed (which is correct) but should be added as an issue addressed for both DD and MT january releases

**Comment by dfoll** [comment]
build 16 was built from dev after the merge

**Comment by dfoll** [comment]
Documentation has been updated on DEV and in the builds for the pdf for this issue if someone wants to verify.

**Comment by BobBooth** [comment]
I see the issue in solutionsdev for DD and MT.  Also in both PDFs. This is verified.
